client/src/views/Dashboard.vue

```diff
@@ -107,138 +141,77 @@ function setEditState(todoId: string, value: boolean) {
   todos.value[indexToUpdateAt].isEditing = value;
 }
 
-function handleCheckboxClick(todoId: number) {
-  const indexToUpdateAt = todos.value.findIndex(
-    (todo) => parseInt(todo._id) === todoId,
-  );
-  const todoToMakeFloat = todos.value[indexToUpdateAt];
-  todoToMakeFloat.isChecked = !todoToMakeFloat.isChecked;
-  if (isSortingApplied.value) {
-    return;
-  } else {
-    setTimeout(() => {
-      if (todoToMakeFloat.isChecked) {
-        todos.value.splice(indexToUpdateAt, 1);
-        todos.value.unshift(todoToMakeFloat);
-      } else {
-        todos.value.splice(indexToUpdateAt, 1);
-        todos.value.push(todoToMakeFloat);
-      }
-    }, 500);
+async function handleCheckboxClick(todoId: string) {
+  try {
+    const updatedTodo = await todoService.updateIsChecked(todoId);
+
+    const indexToUpdateAt = todos.value.findIndex(
+      (todo) => todo._id === todoId,
+    );
+
+    if (indexToUpdateAt === -1)
+      throw new Error('Todo not found in the todos list.');
```

Comment:
please add {} to your if conditions even if it's a one liner, it will help readability if you need to add another functionalities in the future